pkg/controllers/disruption/consolidation_test.go

```diff
@@ -3583,4 +3584,241 @@ var _ = Describe("Consolidation", func() {
 			Expect(result.RequeueAfter).To(BeNumerically(">", 0))
 		})
 	})
+	Context("Metrics", func() {
+		var nodeClaims []*v1beta1.NodeClaim
+		var nodes []*v1.Node
+		var rs *appsv1.ReplicaSet
+		var numNodes int
+		var labels map[string]string
+
+		BeforeEach(func() {
+			numNodes = 25
+			labels = map[string]string{
+				"app": "test",
+			}
+			nodeClaims, nodes = test.NodeClaimsAndNodes(numNodes, v1beta1.NodeClaim{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						v1beta1.NodePoolLabelKey:     nodePool.Name,
+						v1.LabelInstanceTypeStable:   leastExpensiveInstance.Name,
+						v1beta1.CapacityTypeLabelKey: leastExpensiveOffering.CapacityType,
+						v1.LabelTopologyZone:         leastExpensiveOffering.Zone,
+					},
+				},
+				Status: v1beta1.NodeClaimStatus{
+					Allocatable: map[v1.ResourceName]resource.Quantity{
+						v1.ResourceCPU:  resource.MustParse("32"),
+						v1.ResourcePods: resource.MustParse("100"),
+					},
+				}},
+			)
+			// create our RS so we can link a pod to it
+			rs = test.ReplicaSet()
+			ExpectApplied(ctx, env.Client, rs)
+		})
+		It("should update metrics for empty-nodeclaim consolidation", func() {
+			ExpectApplied(ctx, env.Client, rs, nodePool)
+			for _, nodeClaim := range nodeClaims {
+				ExpectApplied(ctx, env.Client, nodeClaim)
+			}
+			for _, node := range nodes {
+				ExpectApplied(ctx, env.Client, node)
+			}
+
+			// inform cluster state about nodes and nodeClaims
+			ExpectMakeNodesAndNodeClaimsInitializedAndStateUpdated(ctx, env.Client, nodeStateController, nodeClaimStateController, nodes, nodeClaims)
+
+			var wg sync.WaitGroup
+			wg.Add(1)
+			finished := atomic.Bool{}
+			go func() {
+				defer GinkgoRecover()
+				defer wg.Done()
+				defer finished.Store(true)
+				ExpectReconcileSucceeded(ctx, disruptionController, client.ObjectKey{})
+			}()
+
+			ExpectTriggerVerifyAction(&wg)
+
+			// controller should finish
+			Eventually(finished.Load, 10*time.Second).Should(BeTrue())
+			wg.Wait()
+
+			// should have no nodeClaims deleted from single nodeClaim consolidation
```

Comment:
I think this comment is wrong, given the fact that all the nodes got deleted here